app/plugins/eu.esdihumboldt.hale.app.transform/src/eu/esdihumboldt/hale/app/transform/ExecApplication.groovy

```diff
@@ -215,6 +221,30 @@ $baseCommand
 					warn('Unexpected parameter -providerId')
 				}
 				break
+
+			case '-filter':
+				executionContext.filters.addUnconditionalFilter(value);
+				break
+
+			case '-filter-on':
+			//<Type>:CQL:<EXPRESSION>
+				if(value.findIndexValues {it = ':'}.size() >=2 ){
+					int typeNameIndex = value.indexOf(':')
+					if(typeNameIndex>0){
+						String typeName = value.take(typeNameIndex)
+						String expression = value.substring(typeNameIndex+1)
+						executionContext.filters.addTypeFilter(typeName, expression)
+					}
+				}else {
+					warn("Illegal parameter -filter-on should be applied with Type")
+				}
+
+				break
+			case '-exclude-type':
+			// <Type1>:<Type2>:<Type3>
```

Comment:
OK, I see now what you hint at in the usage documentation. The colon is used with different semantics in the other places (not as list separator), so I would prefer not to use it here in a different manner. I think it is alright if `-exclude-type` just takes one type as argument.